source/include/core_http_config_defaults.h

```diff
@@ -35,6 +35,24 @@
 #ifndef CORE_HTTP_CONFIG_DEFAULTS_
 #define CORE_HTTP_CONFIG_DEFAULTS_
 
+/* The macro definition for HTTP_DO_NOT_USE_CUSTOM_CONFIG is for Doxygen
+ * documentation only. */
+
+/**
+ * @brief Define this macro to build the HTTP client library without the custom
+ * config file core_http_config.h.
+ *
+ * Without the custom config, the HTTP client library builds with default values
+ * of config macros defined in core_http_config_defaults.h file.
+ *
+ * If a custom config is provided, then HTTP_DO_NOT_USE_CUSTOM_CONFIG should not
+ * be defined.
+ */
+#ifdef DOXYGEN
+    #define HTTP_DO_NOT_USE_CUSTOM_CONFIG
```

Comment:
Dan didn't like it when I did this in coreJSON. Would it be possible to copy the @brief text directly into the doxygen file?

Reply:
Actually if this file is made solely for documentation, then it is probably fine. I think the concern was having this in the library code itself.

Reply:
I have not problem having this in pages.txt only.